flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamingRuntimeContext.java

```diff
@@ -106,17 +109,45 @@ public boolean hasBroadcastVariable(String name) {
 
 	@Override
 	public <T> ValueState<T> getState(ValueStateDescriptor<T> stateProperties) {
-		return operator.getKeyedStateStore().getState(stateProperties);
+		KeyedStateStore keyedStateStore = checkPreconditionsAndGetKeyedStateStore(stateProperties);
+
+		try {
+			stateProperties.initializeSerializerUnlessSet(getExecutionConfig());
+			return keyedStateStore.getState(stateProperties);
+		} catch (Exception e) {
+			throw new RuntimeException("Error while getting state", e);
+		}
 	}
 
 	@Override
 	public <T> ListState<T> getListState(ListStateDescriptor<T> stateProperties) {
-		return operator.getKeyedStateStore().getListState(stateProperties);
+		KeyedStateStore keyedStateStore = checkPreconditionsAndGetKeyedStateStore(stateProperties);
+
+		try {
+			stateProperties.initializeSerializerUnlessSet(getExecutionConfig());
+			return keyedStateStore.getListState(stateProperties);
+		} catch (Exception e) {
+			throw new RuntimeException("Error while getting state", e);
+		}
 	}
 
 	@Override
 	public <T> ReducingState<T> getReducingState(ReducingStateDescriptor<T> stateProperties) {
-		return operator.getKeyedStateStore().getReducingState(stateProperties);
+		KeyedStateStore keyedStateStore = checkPreconditionsAndGetKeyedStateStore(stateProperties);
+
+		try {
+			stateProperties.initializeSerializerUnlessSet(getExecutionConfig());
+			return keyedStateStore.getReducingState(stateProperties);
+		} catch (Exception e) {
+			throw new RuntimeException("Error while getting state", e);
```

Comment:
Do we need to re-wrap the exceptions here? Can we simply let the original exception bubble up?